homeassistant/core.py

```diff
@@ -308,6 +300,18 @@ def async_stop(self) -> None:
 
         self.loop.stop()
 
+    @callback
+    def async_stop_handler(self, *args):
```

Comment:
Oh, `async_stop` already exists. Well, in that case can we move assigning `exit_code` into that method and remove `async_stop_handler` completely?